src/segment/segmentTemplate.js

```diff
@@ -151,7 +152,26 @@ export const segmentsFromTemplate = (attributes, segmentTimeline) => {
     RepresentationID: attributes.id,
     Bandwidth: parseInt(attributes.bandwidth || 0, 10)
   };
-  const mapUri = constructTemplateUrl(attributes.initialization || '', templateValues);
+
+  let mapSegment = { uri: '', resolvedUri: resolveUrl(attributes.baseUrl || '', '') };
+
+  if (attributes.initialization && typeof attributes.initialization === 'string') {
+    const mapUri = constructTemplateUrl(attributes.initialization || '', templateValues);
+
+    mapSegment = {
+      uri: mapUri,
+      resolvedUri: resolveUrl(attributes.baseUrl || '', mapUri)
+    };
+  }
+
+  if (attributes.initialization && typeof attributes.initialization === 'object') {
+    mapSegment = urlTypeToSegment({
+      baseUrl: attributes.baseUrl,
+      source: attributes.initialization.sourceURL,
```

Comment:
With the changes proposed in my other comment to handle the string to object conversion in `getSegmentInformation`, I think you can safely do 

```
source: constructTemplateUrl(attributes.initialization.sourceURL, templateValues)
```

In the case that the `<Initialization>` node took precedence, there shouldn't be any template identifiers, so `constructTemplateUrl` here should just return the `sourceURL` unchanged